src/nacldid.ts

```diff
@@ -375,48 +376,45 @@ export function didToEncPubKey(did: string): Uint8Array {
 /**
  * Create a new NaCLIDentity
  */
-export function createIdentity(): NaCLIdentity {
-  return new NaCLIdentity(nacl.sign.keyPair())
+export function createIdentity(didResolver?: Resolver): NaCLIdentity {
+  return new NaCLIdentity(nacl.sign.keyPair(), didResolver)
 }
 
 /**
  * Instantiates a Serialized NaCLIDentity
  * @param sId 
  */
-export function loadIdentity(sId: SerializableNaCLIdentity): NaCLIdentity {
-  const id = new NaCLIdentity(nacl.sign.keyPair.fromSecretKey(naclutil.decodeBase64(sId.privateKey)))
+export function loadIdentity(sId: SerializableNaCLIdentity, didResolver?: Resolver): NaCLIdentity {
+  const id = new NaCLIdentity(nacl.sign.keyPair.fromSecretKey(naclutil.decodeBase64(sId.privateKey)), didResolver)
   if (id.did !== sId.did) throw new Error('Provided PrivateKey does not match the DID')
   return id
 }
 
 /**
  * Registers `nacl` DID resolver
  */
-export function registerNaclDID() {
-  async function naclDIDResolve(
-    did: string,
-    parsed: ParsedDID,
-  ): Promise<DIDDocument | null> {
-    const publicKey = decodeBase64Url(parsed.id)
-    return {
-      '@context': 'https://w3id.org/did/v1',
-      id: did,
-      publicKey: [{
-        id: `${did}#key1`,
-        type: 'ED25519SignatureVerification',
-        owner: did,
-        publicKeyBase64: naclutil.encodeBase64(publicKey)
-      }, {
-        id: `${did}#key2`,
-        type: 'Curve25519EncryptionPublicKey',
-        owner: did,
-        publicKeyBase64: naclutil.encodeBase64(convertPublicKey(publicKey))
-      }],
-      authentication: [{
-        type: 'ED25519SigningAuthentication',
-        publicKey: `${did}#key1`
-      }]
-    }
+export async function resolver(
+  did: string,
+  parsed: ParsedDID,
+): Promise<DIDDocument | null> {
+  const publicKey = decodeBase64Url(parsed.id)
+  return {
+    '@context': 'https://w3id.org/did/v1',
+    id: did,
+    publicKey: [{
+      id: `${did}#key1`,
+      type: 'ED25519SignatureVerification',
+      owner: did,
+      publicKeyBase64: naclutil.encodeBase64(publicKey)
+    }, {
+      id: `${did}#key2`,
+      type: 'Curve25519EncryptionPublicKey',
```

Comment:
Should we try to register this type of key in https://w3c-ccg.github.io/ld-cryptosuite-registry/
?

Cc @awoie

Reply:
good question. We use it elsewhere and I think others might do so as well